src/utils/conversion.ts

```diff
@@ -2155,9 +2155,8 @@ export function requirementToComplianceRequirement(
       const { claims } = condition;
       conditionContent = claims.map(claim => claimToMeshClaim(claim, context));
     } else if (condition.type === ConditionType.IsIdentity) {
-      const {
-        identity: { did },
-      } = condition;
+      const { identity } = condition;
+      const did = identity instanceof Identity ? identity.did : identity;
```

Comment:
Since the `Condition` type should remain the same, I suggest not making any changes to this function, instead converting the inputs in the procedure from `Identity | string` to `Identity` at the same time that you're checking for existence, and then pass the converted result to this function